### ISSUE TITLE
get_token_accounts_by_owner fails unexpectedly

### ISSUE THREAD

**Issue by Kubikasz** [michaelhly/solana-py#301]
When using the function it just crashes. There are no errors or exception happening. 
What am I doing:
```python
        print("getting token accounts")
        opts = TokenAccountOpts(
            program_id=constants.TOKEN_PROGRAM, encoding="jsonParsed"
        )
        result = None
        try:
            result = self.client.get_token_accounts_by_owner(self.public_key, opts)
        except Exception as e:
            print("error getting token accounts: ", e)
            return None
``` 
and getting empty error in the exception.

**Comment by Kubikasz** [comment]
solved by using `https://solana-api.projectserum.com/` as an endpoint. Still not sure if it is an Solana-py bug or Solana bug

**Comment by pietrodn** [comment]
It appears that in the request body the key `program_id` is replaced with the key `mint`, even if the user specified `program_id` in `TokenAccountOpts`.

**Comment by kevinheavey** [comment]
Closed by #308
